refactor: Replace style prop with utility css classes in frontend/src/lib

Dejvino opened the pull request:
## Problem

Refactoring from style prop to equivalent css classes related to issue #16826.

## Changes

No visual or functional impact.

## Does this work well for both Cloud and self-hosted?

Yes.

## How did you test this code?

Tested through dev tools in storybook, see individual screenshots:

- frontend/src/lib/components/...
	- InsightCard.stories.tsx
		- ![01](https://github.com/PostHog/posthog/assets/1280433/20a59507-64e3-4a2f-99ee-2c0e5c47436a)
	- CommandBar.tsx
		- ![02](https://github.com/PostHog/posthog/assets/1280433/814230b3-77ba-4145-b42d-c9e2a8ce6d29)
	- CompactList.stories.tsx
		-  ![03](https://github.com/PostHog/posthog/assets/1280433/a9af776d-9b73-4607-8ad4-68929f82258d)
	- RollingDateRangeFilter.tsx
		-  ![04](https://github.com/PostHog/posthog/assets/1280433/2fa21a35-58d9-4e24-9155-3a7b8e15f373)
	- HedgehogBuddy.stories.tsx
		-  ![05](https://github.com/PostHog/posthog/assets/1280433/67b9cc78-577c-4864-b043-d3ae4ee96de9)
	- DebugNotice.tsx
		-  ![06](https://github.com/PostHog/posthog/assets/1280433/e1ebe452-fa82-4b31-9cdc-72adb8e03adf)
- frontend/src/lib/lemon-ui/...
	- LemonCalendarSelect.stories.tsx
		-  ![07](https://github.com/PostHog/posthog/assets/1280433/6f96b649-e04d-4e16-959a-98ccab9678a0)
	- LemonCalendarRange.stories.tsx
		-  ![08](https://github.com/PostHog/posthog/assets/1280433/0c1282cd-47da-42cd-a527-e64363c25aa7)

Dejvino committed:
Issue 16826: Replace style prop with utility css classes - frontend/src/lib

daibhin commented on frontend/src/lib/components/CompactList/CompactList.stories.tsx:
```diff
@@ -44,8 +43,7 @@ export function CompactList_({ loading }: { loading: boolean }): JSX.Element {
             </div>
-            {/* eslint-disable-next-line react/forbid-dom-props */}
-            <div style={{ width: 350, marginLeft: 30 }}>
+            <div className="w-[350px] ml-[30px]">
```
```suggestion
            <div className="w-80 ml-8">
```

daibhin commented on frontend/src/lib/components/CommandBar/CommandBar.tsx:
```diff
@@ -21,13 +21,10 @@ const CommandBarOverlay = forwardRef<HTMLDivElement, CommandBarOverlayProps>(fun
-                backdropFilter: 'blur(var(--modal-backdrop-blur))',
-            }}
+            className={`fixed top-0 left-0 w-full h-full flex flex-col items-center justify-center p-3
+                z-[var(--z-command-palette)]
+                background-[var(--modal-backdrop-color)]
+                backdrop-blur-[var(--modal-backdrop-blur)]`}
```
We should not use dynamic class names as per the Tailwind docs: https://tailwindcss.com/docs/content-configuration#dynamic-class-names.

It's probably fine to leave this explicit skip in place or to create a `CommandBar__overlay` class in the associated index.scss file

Dejvino replied:
I'm not sure if this would be considered as dynamic class names (they are defined here statically in whole), but creating the new class looks like a good idea, thanks!

daibhin replied:
Yeah, you're actually correct here. My bad

daibhin commented on frontend/src/lib/components/HedgehogBuddy/HedgehogBuddy.stories.tsx:
```diff
@@ -11,8 +11,7 @@ export default meta
     return (
-        // eslint-disable-next-line react/forbid-dom-props
-        <div style={{ height: 200 }}>
+        <div className="h-[200px]">
```
Let's keep things standard here too:
```suggestion
        <div className="h-52">
```

daibhin commented on frontend/src/lib/lemon-ui/LemonCalendar/LemonCalendarSelect.stories.tsx:
```diff
@@ -23,8 +23,7 @@ const BasicTemplate: StoryFn<typeof LemonCalendarSelect> = (props: LemonCalendar
     return (
-        // eslint-disable-next-line react/forbid-dom-props
-        <div style={{ paddingBottom: 500 }}>
+        <div className="pb-[500px]">
```
Let's use rem her for consistency
```suggestion
        <div className="pb-[32rem]">
```

daibhin commented on frontend/src/lib/lemon-ui/LemonCalendarRange/LemonCalendarRange.stories.tsx:
```diff
@@ -22,8 +22,7 @@ const BasicTemplate: StoryFn<typeof LemonCalendarRange> = (props: LemonCalendarR
     return (
-        // eslint-disable-next-line react/forbid-dom-props
-        <div style={{ paddingBottom: 500 }}>
+        <div className="pb-[500px]">
```
Same here
```suggestion
        <div className="pb-[32rem]">
```

daibhin reviewed (commented):
Left a few comments but thank you for taking this on 🙏 Let me know if there's anything that doesn't make sense - happy to work with you and get this over the line

daibhin reviewed (approved):
Left a few comments but thank you for taking this on 🙏 Let me know if there's anything that doesn't make sense - happy to work with you and get this over the line

daibhin reviewed (changes requested):
Left a few comments but thank you for taking this on 🙏 Let me know if there's anything that doesn't make sense - happy to work with you and get this over the line

Dejvino committed:
Issue 16826: Replace style prop with utility css classes - frontend/src/lib
- use of standardized classes
- extracted CommandBar__overlay class
- px replaced by rem

Dejvino committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into issue-16826-lib

Dejvino commented:
Thank you for the feedback, @daibhin ! Updated PR with the changes.

daibhin commented on frontend/src/lib/components/CommandBar/index.scss:
```diff
@@ -20,3 +20,27 @@
     height: calc(100% - 2.375rem);
 }
+
+.CommandBar__overlay {
+    position: fixed;
+    top: 0;
+    left: 0;
+    z-index: var(--z-command-palette);
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    width: 100%;
+    height: 100%;
+    padding: 0.75rem;
+    background-color: var(--modal-backdrop-color);
+    backdrop-filter: blur(var(--modal-backdrop-blur));
+}
+
+.CommandBar__overlay-content {
+    width: 100%;
+    max-width: 62rem;
+    height: 100%;
+    max-height: 40rem;
+    overflow: hidden;
+}
```
```suggestion
.CommandBar__overlay {
    position: fixed;
    top: 0;
    left: 0;
    z-index: var(--z-command-palette);
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100%;
    padding: 0.75rem;
    background-color: var(--modal-backdrop-color);
    backdrop-filter: blur(var(--modal-backdrop-blur));
    
    &-content {
        width: 100%;
        max-width: 62rem;
        height: 100%;
        max-height: 40rem;
        overflow: hidden;
    }
}
```

Dejvino replied:
Ah, wonderful! 👍🏻

daibhin reviewed (approved):
One small suggestion but nothing blocking! Looks great - thanks for taking this on :)

Dejvino committed:
Update frontend/src/lib/components/CommandBar/index.scss

- use of parent selector

Co-authored-by: David Newell <d.newell1@outlook.com>

Dejvino committed:
Merge branch 'master' into issue-16826-lib

Dejvino committed:
Merge branch 'refs/heads/master' into issue-16826-lib

Dejvino committed:
Prettier fix of frontend/src/lib/components/CommandBar/index.scss

Dejvino committed:
Merge branch 'refs/heads/master' into issue-16826-lib

Dejvino committed:
Merge remote-tracking branch 'refs/remotes/upstream/master' into issue-16826-lib

Dejvino commented:
Could I get a workflows re-run? 🙏🏻 
@daibhin

daibhin committed:
Merge branch 'master' into issue-16826-lib

Dejvino committed:
Merge remote-tracking branch 'refs/remotes/upstream/master' into issue-16826-lib

daibhin commented:
@Dejvino you might need to pull the latest `master` and merge it into this branch for those tests to pass

daibhin commented:
Hey @Dejvino - is this something you would like me to take over for you?

Dejvino commented:
@daibhin not yet. Sorry for the delay, holidays happened 🙂
I think I'm missing some snapshots updates to pass the tests, I'm working on that now.

Dejvino committed:
Merge branch 'master' into issue-16826-lib

Dejvino committed:
Merge branch 'master' into issue-16826-lib

Dejvino committed:
Fix CommandBar snapshot tests by adding fixed class

Dejvino committed:
Update snapshots for CompactList

Dejvino committed:
Update snapshots for LemonCalendar

Dejvino commented:
Snapshots should now be updated.

FWIW this took more time than I expected due to #23699.

Dejvino committed:
Merge branch 'master' into issue-16826-lib

Dejvino committed:
Merge branch 'master' into issue-16826-lib

Dejvino commented:
Hi @daibhin , I expect the last workflows failure (`Error: You must pass your PostHog project's api key.`)  is unrelated and should be solved with the pull from `master`?

Dejvino committed:
Merge branch 'master' into issue-16826-lib

Dejvino commented:
Ok, that didn't help. Looks like the API token is not getting passed correctly (`posthog-token: ${{secrets.POSTHOG_API_TOKEN}}`).

I'm afraid there is not much more I can do here @daibhin

daibhin commented:
Don't know what is up here @Dejvino. I just cherry picked your changes into https://github.com/PostHog/posthog/pull/24293 which should hopefully work

daibhin commented:
I just merged https://github.com/PostHog/posthog/pull/24293. Thank you @Dejvino for all your help on this 🙏
